### PR TITLE
added sorting to opportunities; 

### PR DESCRIPTION
**Pull request by techsmyth**
added sorting to returned aspects
version bump;

### CODE REVIEW

File: src/domain/challenge/challenge/challenge.service.ts [hunk 1]
```diff
@@ -386,7 +386,14 @@ export class ChallengeService {
       `Querying all Opportunities with limit: ${limit} and shuffle: ${shuffle}`,
       LogContext.CHALLENGES
     );
-    return limitAndShuffle(opportunities, limit, shuffle);
+
+    const limitAndShuffled = limitAndShuffle(opportunities, limit, shuffle);
+
+    // Sort the opportunities base on their display name
+    const sortedOpportunities = limitAndShuffled.sort((a, b) =>
+      a.displayName > b.displayName ? 1 : -1
```
Comment (me-andre) [inline src/domain/challenge/challenge/challenge.service.ts:394]: In that case, you can just `sort()` without the callback. In JS it's alphabetical by default.

File: src/domain/context/context/context.service.ts [hunk 2]
```diff
@@ -337,8 +337,15 @@ export class ContextService {
...
     if (!aspectIDs) {
-      const aspects = limitAndShuffle(contextLoaded.aspects, limit, shuffle);
-      return aspects;
+      const limitAndShuffled = limitAndShuffle(
+        contextLoaded.aspects,
+        limit,
+        shuffle
+      );
+      const sortedAspects = limitAndShuffled.sort((a, b) =>
+        a.displayName > b.displayName ? 1 : -1
```
Comment (me-andre) [inline src/domain/context/context/context.service.ts:346]: https://github.com/alkem-io/server/pull/1877/files#r850528263

**Comment by Comoque1** [comment]
Tested on Test env with this branch.

Opportunities shuffling don't apply on Challenge dashboard page. The order is fine on Opportunities list.
image

File: src/domain/challenge/challenge/challenge.service.ts (hunk 1 above)
Comment (techsmyth) [reply src/domain/challenge/challenge/challenge.service.ts:394]: how does it know then to pick up the displayName field?

Comment (me-andre) [reply src/domain/challenge/challenge/challenge.service.ts:394]: Oh sorry, missed that

**Review by me-andre — APPROVED** [review]

**Pull request merged**